modules/pricer/src/main/java/com/opengamma/strata/pricer/calibration/CalibrationMeasures.java

```diff
@@ -0,0 +1,162 @@
+   * 
+   * @param measures  the list of measures
+   * @return the calibration measures
+   * @throws IllegalArgumentException if a trade type is specified more than once
+   */
+  @SafeVarargs
+  public static CalibrationMeasures of(CalibrationMeasure<? extends Trade>... measures) {
+    return new CalibrationMeasures(ImmutableList.copyOf(measures));
+  }
+
+  //-------------------------------------------------------------------------
+  // restricted constructor
+  private CalibrationMeasures(List<? extends CalibrationMeasure<? extends Trade>> measures) {
+    ImmutableMap.Builder<Class<?>, CalibrationMeasure<? extends Trade>> builder = ImmutableMap.builder();
+    for (CalibrationMeasure<? extends Trade> measure : measures) {
+      builder.put(measure.getTradeType(), measure);
+    }
+    measuresByTrade = builder.build();
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the supported trade types.
+   * 
+   * @return the supported trade types
+   */
+  public ImmutableSet<Class<?>> getTradeTypes() {
+    return measuresByTrade.keySet();
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the value, such as par spread.
+   * <p>
+   * The value must be calculated using the specified curve provider.
+   * 
+   * @param trade  the trade
+   * @param provider  the rates provider
+   * @return the sensitivity
+   * @throws IllegalArgumentException if the trade cannot be valued
+   */
+  public double value(Trade trade, RatesProvider provider) {
+    CalibrationMeasure<Trade> measure = getMeasure(trade.getClass());
+    return measure.value(trade, provider);
+  }
+
+  /**
+   * Calculates the sensitivity with respect to the curve provider as a long vector.
+   * <p>
+   * The vector is composed of the concatenated curve sensitivities.
+   * 
+   * @param trade  the trade
+   * @param provider  the rates provider
+   * @param curveOrder  the order of the curves
+   * @return the sensitivity derivative
+   */
+  public double[] derivative(Trade trade, RatesProvider provider, List<CurveParameterSize> curveOrder) {
+    // determine the curve parameter sensitivities
+    CalibrationMeasure<Trade> measure = getMeasure(trade.getClass());
+    CurveCurrencyParameterSensitivities paramSens = measure.sensitivities(trade, provider);
+
+    // ignore the currency
+    CurveUnitParameterSensitivities unitSens = CurveUnitParameterSensitivities.empty();
+    for (CurveCurrencyParameterSensitivity ccySens : paramSens.getSensitivities()) {
+      unitSens = unitSens.combinedWith(CurveUnitParameterSensitivity.of(ccySens.getMetadata(), ccySens.getSensitivity()));
+    }
+
+    // expand to a concatenated array
+    double[] result = DoubleArrayMath.EMPTY_DOUBLE_ARRAY;
+    for (CurveParameterSize curveParams : curveOrder) {
+      double[] sens = unitSens.findSensitivity(curveParams.getName())
+          .map(s -> s.getSensitivity())
```

Comment:
I have mixed feelings. In this case I'm inclined to agree. I think

    .map(item -> item.foo())

is probably clearer than

    .map(ItemClass::foo)

but

    .map(this::bar)

is nicer than

    .map(item -> bar(item))

Having said that, I don't have a strong opinion either way. I tend to use method references because IntelliJ underlines expressions that could be converted and offers to convert them for me.